examples/platforms/nrf52840/diag.c

```diff
@@ -191,16 +227,75 @@ static void processTransmit(otInstance *aInstance, int argc, char *argv[], char
         snprintf(aOutput, aOutputMaxLen, "set diagnostic messages count to %" PRId32 "\r\nstatus 0x%02x\r\n", sTxRequestedCount,
                  error);
     }
+    else
+    {
+        error = OT_ERROR_INVALID_ARGS;
+    }
+
+exit:
+    appendErrorResult(error, aOutput, aOutputMaxLen);
+}
+
+static void processTemp(otInstance *aInstance, int argc, char *argv[], char *aOutput, size_t aOutputMaxLen)
+{
+    OT_UNUSED_VARIABLE(aInstance);
+    OT_UNUSED_VARIABLE(argv);
+
+    otError error = OT_ERROR_NONE;
+
+    otEXPECT_ACTION(otPlatDiagModeGet(), error = OT_ERROR_INVALID_STATE);
+    otEXPECT_ACTION(argc == 0, error = OT_ERROR_INVALID_ARGS);
+
+    int32_t temperature = nrf5TempGet();
+
+    snprintf(aOutput, aOutputMaxLen, "%" PRId32 ".%02" PRId32 "\r\n", temperature / 4, 25 * (temperature % 4));
```

Comment:
A comment describing the units that this string is presenting would be helpful.  Clearly there is some formula that allowed you to translate between the register value and what I presume is degrees C.